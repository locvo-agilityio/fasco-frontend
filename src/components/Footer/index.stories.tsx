import type { Meta, StoryObj } from '@storybook/react';

// Components
import Footer from '.';

const meta = {
  title: 'Components/Footer',
  component: Footer,
  parameters: {
    layout: 'centered',
  },
  tags: ['autodocs'],

  render: () => (
    <div className="w-[1280px]">
      <Footer />
    </div>
  ),
} satisfies Meta<typeof Footer>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {};
