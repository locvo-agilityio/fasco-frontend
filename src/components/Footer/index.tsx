'use client';

import { useMemo } from 'react';
import {
  Navbar as NavbarHeroUI,
  NavbarBrand,
  NavbarContent,
  NavbarItem,
} from '@heroui/react';
import Link from 'next/link';
import { usePathname } from 'next/navigation';

// Constants
import { FOOTER_NAVBAR } from '@/constants';

const Footer = () => {
  const pathname = usePathname();

  const renderNavbar = useMemo(
    () =>
      FOOTER_NAVBAR.map(({ id, title, path }) => (
        <NavbarItem
          key={id}
          className="font-poppins text-foreground-400 hover:border-b-2 hover:border-b-primary-200 data-[active=true]:border-b-2 data-[active=true]:border-b-primary-200 data-[active=true]:font-normal"
          isActive={pathname === path}
        >
          <Link href={path}>{title}</Link>
        </NavbarItem>
      )),
    [pathname],
  );

  return (
    <NavbarHeroUI
      className="py-8 border-t mt-10 border-t-primary-200"
      maxWidth="2xl"
    >
      <NavbarBrand>
        <h2 className="text-2xl font-volkhov">FASCO</h2>
      </NavbarBrand>

      <div className="hidden sm:flex gap-14 items-center">
        <NavbarContent className="hidden sm:flex gap-14" justify="center">
          {renderNavbar}
        </NavbarContent>
      </div>
    </NavbarHeroUI>
  );
};

export default Footer;
