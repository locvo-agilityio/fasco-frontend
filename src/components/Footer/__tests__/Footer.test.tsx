import { render } from '@testing-library/react';

// Components
import Footer from '..';

describe('Footer', () => {
  it('should render correctly', () => {
    const { container } = render(<Footer />);

    expect(container).toMatchSnapshot();
  });
});
