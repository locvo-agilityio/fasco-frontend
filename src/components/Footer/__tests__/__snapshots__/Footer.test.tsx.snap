// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should render correctly 1`] = `
<div>
  <nav
    class="flex z-40 w-full h-auto items-center justify-center data-[menu-open=true]:border-none sticky top-0 inset-x-0 backdrop-blur-lg data-[menu-open=true]:backdrop-blur-xl backdrop-saturate-150 bg-background/70 py-8 border-t mt-10 border-t-primary-200"
    style="--navbar-height: 4rem;"
  >
    <header
      class="z-40 flex px-6 gap-4 w-full flex-row relative flex-nowrap items-center justify-between h-[var(--navbar-height)] max-w-[1536px]"
    >
      <div
        class="flex basis-0 flex-row flex-grow flex-nowrap justify-start bg-transparent items-center no-underline text-medium whitespace-nowrap box-border"
      >
        <h2
          class="text-2xl font-volkhov"
        >
          FASCO
        </h2>
      </div>
      <div
        class="hidden sm:flex gap-14 items-center"
      >
        <ul
          class="h-full flex-row flex-nowrap items-center data-[justify=start]:justify-start data-[justify=start]:flex-grow data-[justify=start]:basis-0 data-[justify=center]:justify-center data-[justify=end]:justify-end data-[justify=end]:flex-grow data-[justify=end]:basis-0 hidden sm:flex gap-14"
          data-justify="center"
        >
          <li
            class="text-medium whitespace-nowrap box-border list-none font-poppins text-foreground-400 hover:border-b-2 hover:border-b-primary-200 data-[active=true]:border-b-2 data-[active=true]:border-b-primary-200 data-[active=true]:font-normal"
          >
            <a
              href="/support-center"
            >
              Support Center
            </a>
          </li>
          <li
            class="text-medium whitespace-nowrap box-border list-none font-poppins text-foreground-400 hover:border-b-2 hover:border-b-primary-200 data-[active=true]:border-b-2 data-[active=true]:border-b-primary-200 data-[active=true]:font-normal"
          >
            <a
              href="/invoicing"
            >
              Invoicing
            </a>
          </li>
          <li
            class="text-medium whitespace-nowrap box-border list-none font-poppins text-foreground-400 hover:border-b-2 hover:border-b-primary-200 data-[active=true]:border-b-2 data-[active=true]:border-b-primary-200 data-[active=true]:font-normal"
          >
            <a
              href="/contract"
            >
              Contract
            </a>
          </li>
          <li
            class="text-medium whitespace-nowrap box-border list-none font-poppins text-foreground-400 hover:border-b-2 hover:border-b-primary-200 data-[active=true]:border-b-2 data-[active=true]:border-b-primary-200 data-[active=true]:font-normal"
          >
            <a
              href="/careers"
            >
              Careers
            </a>
          </li>
          <li
            class="text-medium whitespace-nowrap box-border list-none font-poppins text-foreground-400 hover:border-b-2 hover:border-b-primary-200 data-[active=true]:border-b-2 data-[active=true]:border-b-primary-200 data-[active=true]:font-normal"
          >
            <a
              href="/blog"
            >
              Blog
            </a>
          </li>
          <li
            class="text-medium whitespace-nowrap box-border list-none font-poppins text-foreground-400 hover:border-b-2 hover:border-b-primary-200 data-[active=true]:border-b-2 data-[active=true]:border-b-primary-200 data-[active=true]:font-normal"
          >
            <a
              href="/faq"
            >
              FAQ,s
            </a>
          </li>
        </ul>
      </div>
    </header>
  </nav>
</div>
`;
